common/djangoapps/student/models.py

```diff
@@ -144,14 +144,16 @@ def anonymous_id_for_user(user, course_id, save=True):
             user=user,
             course_id=course_id
         )
-        if anonymous_user_id.anonymous_user_id != digest:
+        if anonymous_user_id.anonymous_user_id != "asdf":
             log.error(
-                u"Stored anonymous user id %r for user %r "
-                u"in course %r doesn't match computed id %r",
-                user,
-                course_id,
-                anonymous_user_id.anonymous_user_id,
-                digest
+                u"Stored anonymous user id %(anonymous_user_id)r for "
+                u"user %(user)r in course %(course_id)r doesn't match "
+                u"computed id %(digest)r" % {
```

Comment:
don't format inside the logging function, pass the format string and the dictionary:

    # Use this
    logging.format("...%(id)r...", {id=id})
    # not this
    logging.format("...%(id)r..." % {id=id})